### DISCUSSION TITLE
SD card on Nucleo F411RE not working

### DISCUSSION THREAD

**Discussion by B00mb0x** [micropython/micropython#11957] [STM32 / Pyboard]
I am currently using a Nucleo F411RE Board.
I flashed the board with the micropython commit 813d559 from June 19th 2023 using the command `make BOARD=NUCLEO_F411RE deploy-stlink `.
Micropython works fine, but I want to import the SDCard module from machine using:
`from machine import SDCard`
I get an error message:
`Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: can't import name SDCard`

While building, it showed also bulding  'sdcard.c'. I don't know, why it still can't be imported.
![Screenshot_20230706_143702](https://github.com/micropython/micropython/assets/20295167/00b2311b-a396-4a9b-8705-e6b0a22e39f5)

Is it possible to operate an SD card from a Nucleo F411RE? 
How can I achieve it? Should I change the makefile?
Or did I just try to import the wrong thing?

Thanks in advance!

**Comment by rkompass** [comment]
I once had the same problem. Could be that sdcard.c is for special sdcard peripherals that the F411 does not have. 
It might be compiled but not linked in.
In my case (with the board definition by @mcauser for the blackpill board [here](https://github.com/mcauser/WEACT_F411CEU6)) the python driver in [micropython-lib](https://github.com/micropython/micropython-lib/blob/master/micropython/drivers/storage/sdcard/sdcard.py) worked. You may compile it with mpy-cross and have it in the file system or freeze it in.
